zanata-war/src/main/java/org/zanata/service/impl/TranslationMemoryServiceImpl.java

```diff
@@ -359,9 +359,16 @@ private void processIndexMatch(TransMemoryQuery transMemoryQuery,
                     Lists.newArrayList(textFlowTarget.getContents());
             TransMemoryResultItem.MatchType matchType =
                     fromContentState(textFlowTarget.getState());
-            TransMemoryResultItem item = createOrGetResultItem(transMemoryQuery, matchesMap, match, matchType,
-                    textFlowContents, targetContents);
-            addTextFlowTargetToResultMatches(textFlowTarget, item);
+
+            double percent =
+                calculateSimilarityPercentage(transMemoryQuery,
+                    textFlowContents);
+            if (percent != 0) {
```

Comment:
I think it include 0% TM is because the actual result is of type double. So a 0.005 (0.5%) match is either truncated or rounded to 0%. Here you may need to check if the percent is too small instead. e.g. less than 0.01 or 0.05